### DISCUSSION TITLE
Limit local cache size

### DISCUSSION THREAD

**Discussion by maxburs** [vercel/turborepo#1264] [Ideas]
I'm caching the /node_modules/.turbo folder in CI. This is causing problems because of its size. I've seen it get as large at 4.5gb. Is there a good way to limit the size of the cache?

**Comment by cobacious** [comment]
We’ve seen it get well over 100GB. Would be nice to have a configuration option to limit its size.

**Reply by cobacious** [reply]
To clarify, that is on local developer machines rather than CI.

**Comment by joshmedeski** [comment]
It would be great to auto-clean when the cache folder gets to a certain size or at the least a notification that warns you when your local cache has hit a certain size so you can manually remove some of the local cache.

Furthermore, a `turbo clean` type command would be useful to remove the local cache automatically instead of manually going into the `node_modules/.cache/turbo`.

**Comment by cobacious** [comment]
Any update from Vercel on this? We're looking to switch to remote caching with a custom server and will need an alternative solution to mitigate the bloat if there's no built in support.

**Comment by mehulkar** [comment]
Thanks for opening. There is no way to do this right now, but I've turned this into a feature request in Issues: #7029

**Comment by mehulkar** [comment]
Note: please upvote #7029 if you're looking at this!
